clients/notification-pubsub/src/main/java/org/eclipse/hono/client/notification/pubsub/PubSubBasedNotificationReceiver.java

```diff
@@ -111,20 +111,17 @@ public Future<Void> stop() {
 
     @SuppressWarnings({ "unchecked", "rawtypes" })
     private void handleMessage(final PubsubMessage message) {
-        PubSubMessageHelper.getPayload(message)
-                .ifPresentOrElse(payload -> {
-                    final Buffer buffer = Buffer.buffer(payload);
-                    final JsonObject json = buffer.toJsonObject();
+        final Buffer buffer = Buffer.buffer(PubSubMessageHelper.getPayload(message));
+        final JsonObject json = buffer.toJsonObject();
```

Comment:
have you checked what happens here when the buffer is empty?

Reply:
checked it .. when the buffer is empty, an exception is thrown, I added a try-catch to catch the exception